### PR TITLE
fix for Unittest discovery not working on Python 2.7

### PR DESCRIPTION
**Pull request by bschnurr**
fix bug #5462

### CODE REVIEW

File: Python/Product/TestAdapter.Executor/PythonFiles/testing_tools/adapter/unittest/_discovery.py [hunk 1]
```diff
@@ -40,15 +40,16 @@ def discover(pytestargs=None, hidestdio=False,
...
                     obj = module
                     for part in parts:
                         try:
                             parent, obj = obj, getattr(obj, part)
                         except AttributeError as e:
                             pass
 
-                    if isinstance(obj, types.FunctionType):
+                    if (isinstance(obj, types.FunctionType) or
+                        isinstance(obj, types.UnboundMethodType)):
```
Comment (bschnurr) [inline Python/Product/TestAdapter.Executor/PythonFiles/testing_tools/adapter/unittest/_discovery.py]: apparently looking at loadTestsFromName from 2.7 python the type on test function is UnboundedMethod

https://github.com/python/cpython/blob/2.7/Lib/unittest/loader.py#L59

Comment (huguesv) [reply Python/Product/TestAdapter.Executor/PythonFiles/testing_tools/adapter/unittest/_discovery.py]: `UnboundMethodType` will fail on Python 3. You should check it's (Python 2 and `UnboundMethodType`).

```
C:\Users\huvalo>py -3
Python 3.8.0a4 (v3.8.0a4:c1004b8546, May  7 2019, 13:53:12) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import types
>>> types.UnboundMethodType
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'types' has no attribute 'UnboundMethodType'
>>> quit()

C:\Users\huvalo>py -2.7
Python 2.7.16 (v2.7.16:413a49145e, Mar  4 2019, 01:37:19) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import types
>>> types.UnboundMethodType
<type 'instancemethod'>
>>>
```

**Comment by huguesv** [comment]
Please make sure to run the TestDiscovererTests unit tests for both 2.7 and 3.7.

`TestDiscovererTests27.DiscoverUnittest` used to fail because of this bug, so it should now pass. Well, all tests should be green or yellow.

File: Python/Product/TestAdapter.Executor/PythonFiles/testing_tools/adapter/unittest/_discovery.py (hunk 1 above)
Comment (bschnurr) [reply Python/Product/TestAdapter.Executor/PythonFiles/testing_tools/adapter/unittest/_discovery.py]: ah good catch

**Pull request merged**